client/dhcp_client_lwt.ml

```diff
@@ -0,0 +1,90 @@
+let src = Logs.Src.create "dhcp_client_lwt"
+module Log = (val Logs.src_log src : Logs.LOG)
+
+module Make(Time : Mirage_time_lwt.S) (Net : Mirage_net_lwt.S) = struct
+  open Lwt.Infix
+
+  type lease = Dhcp_wire.pkt
+
+  type t = lease Lwt_stream.t
+
+  let connect ?(with_xid)
+              ?(requests : Dhcp_wire.option_code list option) net =
+    (* listener needs to occasionally check to see whether the state has advanced,
+     * and if not, start a new attempt at a lease transaction *)
+    let sleep_interval = Duration.of_sec 3 in
+
+    let (client, dhcpdiscover) = Dhcp_client.create ?with_xid ?requests (Net.mac net) in
+    let c = ref client in
+
+    let rec renew c t =
+      Time.sleep_ns @@ Duration.of_sec t >>= fun () ->
+      match Dhcp_client.renew c with
+      | `Noop -> Log.debug (fun f -> f "Can't renew this lease; won't try");  Lwt.return_unit
+      | `Response (c, buf) ->
+        Log.debug (fun f -> f "attempted to renew lease: %a" Dhcp_client.pp c);
+        Net.write net buf >>= function
+          | Error e ->
+            Log.err (fun f -> f "Failed to write lease renewal request: %a" Net.pp_error e);
+            Lwt.return_unit
+          | Ok () ->
+            renew c t (* ideally t would come from the new lease... *)
+    in
+    let rec get_lease push dhcpdiscover =
```

Comment:
no need to pass `push` here (only used for a self-call of `get_lease at the end of this function AFAICS)